front-end-tools/src/writing_to_blockchain.ts

```diff
@@ -20,97 +21,183 @@ export async function deploy(connection: WalletConnection, account: string, base
     } as DeployModulePayload);
 }
 
+export async function write(
+    connection: WalletConnection | undefined,
+    account: string | undefined,
+    inputParameter: string | undefined,
+    contractName: string | undefined,
```

Comment:
Same comments as for `read`.

Not sure why all these `|undefined` are needed in this function. I'd rather if they are handled in the form validation already.

Reply:
see comment above.